lib/dialects/mysql/index.js

```diff
@@ -91,7 +91,7 @@ function connectionErrorHandler(client, connection, err) {
   if (connection && err && err.fatal) {
     if (connection.__knex__disposed) return;
     connection.__knex__disposed = true;
-    client.pool.genericPool.destroy(connection);
+    client.pool.destroy(connection);
```

Comment:
Ah, nevermind.  This happens on error; the others happen when they're done with the connection.